pyprof/nvtx/nvmarker.py

```diff
@@ -68,58 +70,141 @@ def isfunc(mod, f):
     return ins.ismethod(attr) or ins.isfunction(attr) or ins.ismethoddescriptor(attr) or ins.isbuiltin(attr)
 
 
+# Nested dicts of this run's frame names to help uniquify them
+#
+# func_map[(partial_func_stack,frame_name)][filename+lineno] = frame_name_to_use
+#
+func_map = {}
+
 # Returns a dict string with a tracemarker and function stack in it
 #
-def traceMarker():
+def traceMarker(op_name):
+    global func_map
 
     # Return True if the name in the hierarchy should be skipped
-    def should_skip_frame_name(name):
-        if name in ["__call__","wrapper_func","<module>", "always_benchmark_wrapper"]:
+    #
+    def should_skip_frame_name(name, prev_name):
+        # __call__:  
+        #    Much of Torch library is implemented in this way. Ignore these extra layers
+        # wrapper_func and always_benchmark_warpper: 
+        #    Are functions in this file. If there are nested monkeypatched functions 
+        #    we don't want it to show up
+        # <*>: 
+        #    Things like <module>, <genexpr>, <lamba> which don't add any information 
+        #    and break html
+        #
+        for prefix in ["__call__", "wrapper_func", "always_benchmark_wrapper"]:
+            if name.startswith(prefix):
+                return True
+        if name.startswith("<") and name.endswith(">"):
             return True
-        elif name.startswith("<") and name.endswith(">"):
+        if name==prev_name:
             return True
-        else:
-             return False
+        return False
+
+    # Given a function stack, clean it up to remove unwanted fields as 
+    # well as removing any back-to-back duplicates
+    # 
+    def cleanup_func_stack(func_stack, op_name):
 
-    # Returns a string representing the stack of function calls separated with '/'
+        ret = ""
+        prev_fn_name = ""
+        suffix = ""
+
+        x = func_stack.split("/")
+        for fn_name in x:
+
+            # This is used to detect when the same torch op was called 
+            # multiple times from the same parent function. Capture the
+            # count as a 'suffix' and put it on the end of the op name
```

Comment:
Can you provide a code example here?  
Right now, this looks like it just duplicates the code in `should_skip_frame_name`

Reply:
done